src/AspNetCore/Src/inprocessapplication.cxx

```diff
@@ -132,6 +134,125 @@ IN_PROCESS_APPLICATION::FindDotNetFolders(
 }
 
 VOID
+IN_PROCESS_APPLICATION::SetStdOut(
+    VOID
+)
+{
+    HRESULT      hr = S_OK;
+    BOOL         fLocked = FALSE;
+    BOOL         fResult = FALSE;
+
+    if (!m_fDoneStdRedirect)
+    {
+        // Have not set stdout yet, redirect stdout to log file
+        AcquireSRWLockExclusive(&m_srwLock);
+        fLocked = TRUE;
+        if (!m_fDoneStdRedirect)
+        {
+            //
+            // best effort
+            // no need to capture the error code as nothing we can do here
+            // in case mamanged layer exits abnormally, may not be able to capture the log content as it is buffered.
+            //
+            if (!GetConsoleWindow())
+            {
+                hr = m_pConfiguration->CreateLogFile(TRUE, &m_struLogFilePath, &m_hLogFileHandle);
+                if (FAILED(hr))
+                {
+                    goto Finished;
+                }
+                //
+                // SetStdHandle works as w3wp does not have Console
+                // Current process does not have a console
+                //
+                SetStdHandle(STD_ERROR_HANDLE, m_hLogFileHandle);
+                if (m_pConfiguration->QueryStdoutLogEnabled())
+                {
+                    SetStdHandle(STD_OUTPUT_HANDLE, m_hLogFileHandle);
```

Comment:
These comments lead me to think that SetStdHandle doesn't work, right?

Reply:
No. SetStdHandle works for w3wp scenario.

Reply:
Should we remove the commented out code though?

Reply:
no. This is just comment is for maintenance purpose saying this method won't work. No need to try them later.

Reply:
Let's move the comment before the call to SetStdHandle. Also change the comment to say "These other options do not work on azure or don't capture logging early enough"